src/main/scala/nl.knaw.dans.easy.solr4files/components/AuthInfoItem.scala

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files.components
+
+import java.nio.file.{ Path, Paths }
+import java.text.SimpleDateFormat
+import java.util.UUID
+
+import nl.knaw.dans.easy.solr4files._
+import org.joda.time.DateTime
+import org.json4s.ext.{ EnumNameSerializer, JodaTimeSerializers }
+import org.json4s.native.JsonMethods.parse
+import org.json4s.{ DefaultFormats, Formats }
+
+import scala.util.{ Failure, Try }
+
+/** The class arguments must match the json fields returned by the service easy-auth-info */
+case class AuthInfoItem(itemId: String,
+                        owner: String,
+                        dateAvailable: DateTime,
+                        accessibleTo: RightsFor.Value,
+                        visibleTo: RightsFor.Value
+                       ) {
+  val path: Path = Paths.get(itemId.replaceAll("^[^/]+/", ""))
+  val bagID: UUID = UUID.fromString(itemId.replaceAll("/.*", ""))
```

Comment:
Ok, maybe there is a way to not make me wonder how this works?
Seems that you remote the things you don't want.

Reply:
I have to confess that my regex knowledge is not particularly great either. So maybe we should have a convention to add comments for every regex we write. Another solution is to study them and get familiar with them 😜